textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcher.java

```diff
@@ -23,7 +23,7 @@
  * @author laishuying
  */
 public class RegexMatcher implements IOperator {
-    private final IPredicate predicate;
+    private final RegexPredicate regexPredicate;
```

Comment:
There's my thoughts: we need to cast IPrediate to RegexPredicate anyway. So instead of having a IPredicate member variable and cast it in getNextTuple() function, I feel having a RegexPredicate member variable and casting it in the constructor is more explicit. And I'm wondering what do we need to do if the IPredicate is not a RegexPredicate?